osidb/models.py

```diff
@@ -1788,6 +1789,55 @@ class Meta:
         )
 
 
+class FlawReferenceManager(ACLMixinManager, TrackingMixinManager):
+    """flawreference manager"""
+
+    @staticmethod
+    def create_flawreference(flaw, _type, url, **extra_fields):
+        """return a new flawreference or update an existing flawreference without saving"""
+        try:
+            flawreference = FlawReference.objects.get(flaw=flaw, type=_type, url=url)
+            for attr, value in extra_fields.items():
+                setattr(flawreference, attr, value)
+            return flawreference
+
+        except ObjectDoesNotExist:
+            return FlawReference(flaw=flaw, type=_type, url=url, **extra_fields)
+
+
+class FlawReference(TrackingMixin, ACLMixin):
+    """Model representing flaw references"""
+
+    class FlawReferenceType(models.TextChoices):
+        """allowable references"""
+
+        VULN_RESPONSE = "VULN_RESPONSE"
+        EXTERNAL = "EXTERNAL"
+
+    # internal primary key
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    type = models.CharField(
+        choices=FlawReferenceType.choices,
+        default=FlawReferenceType.EXTERNAL,
+        max_length=50,
+    )
+
+    url = models.CharField(max_length=200)
```

Comment:
make it a `TextField`, see https://stackoverflow.com/a/417184

Reply:
we might also want to provide field validation to make sure that it looks like a URL, but that depends on the existing data

Reply:
A validation is a good point. I can use `URLField`, which is already validated by `URLValidator`.